### ISSUE TITLE
cAdvisor is failing to collect the data for processes running inside the container

### ISSUE THREAD

**Issue by Siddharthav1** [google/cadvisor#2858]
Hello,
I am using the cadvisor to collect the container health data.Here, I am using the below command to run the cadvisor:
   
`sudo docker run --volume=/:/rootfs:ro --volume=/var/run:/var/run:ro --volume=/sys:/sys:ro --volume=/var/lib/docker/:/var/lib/docker:ro --volume=/dev/disk/:/dev/disk:ro --publish=8080:8080 --detach=true --name=cadvisor google/cadvisor:latest`

But, the issue is I am getting the CPU and memory related data but it is faillig to collect the state of the process running inside the LXC container.

For example:
I want to get the state of the process **"p"**, it is running inside the LXC container but cadvisor is showing it as running.

ps output inside the container:

```
sh-4.4# ps ax | grep Audio
    105 ?        Ssl    0:00 Audio -d
    364 pts/0    S+     0:00 grep Audio

```
But, in cadvisor dashboard it is not showing as running state. It is giving "0" for the process which is running inside the container:

`container_tasks_state{id="/lxc.payload.mycontainer/system.slice/Audio.service",image="",name="",state="running"} 0`

Can anybody suggect me what wrong I am doing here please.

Regards,
Siddhartha V

**Comment by samcro1967** [comment]
Seeing the same issue...

**Comment by iwankgb** [comment]
I have no experience with LXC at all but quick code analysis suggests that you need to set [`--enable_load_reader` flag](https://github.com/google/cadvisor/blob/bfe982b346633ca5e87021019e9e2ab75aa0d7fe/manager/container.go#L48).

**Comment by Jaxmetalmax** [comment]
> I have no experience with LXC at all but quick code analysis suggests that you need to set [`-enable_load_reader` flag](https://github.com/google/cadvisor/blob/bfe982b346633ca5e87021019e9e2ab75aa0d7fe/manager/container.go#L48).

Even with this flag enabled, the process is not showing up
![Captura de pantalla de 2021-05-11 12-13-55](https://user-images.githubusercontent.com/1606476/117864764-649ceb80-b252-11eb-9d53-e6fd59b69869.png)

and I'm not running with LXC but I've seen others with similar problem

This is what I'm running:
```
sudo docker run   --volume=/:/rootfs:ro   --volume=/var/run:/var/run:ro   --volume=/sys:/sys:ro   --volume=/var/lib/docker/:/var/lib/docker:ro   --volume=/dev/disk/:/dev/disk:ro   --publish=8881:8080   --detach=true   --name=cadvisor   --privileged --pid=host --device=/dev/kmsg   gcr.io/cadvisor/cadvisor:latest --disable_metrics=udp --enable_load_reader=true ```

**Comment by Siddharthav1** [comment]
Hello,

Can I get any inputs on collecting the process data of the container. Nothing is working for me .Kindly suggest me anything please.

regards,
Siddhartha V

**Comment by Siddharthav1** [comment]
hello @Jaxmetalmax ,

on which port you are acessing the cadvisor? I tried at localhost:8080. but getting "Unable to Connect error.". Earlier I used to connect the same port which was accessible,

**Comment by Siddharthav1** [comment]
yes, as @Jaxmetalmax said, even if we use the flag, processes are not visible.
![image](https://user-images.githubusercontent.com/59814967/119520699-24805180-bd98-11eb-856f-4f95bbc0bac1.png)

regards,
Siddhartha V

**Comment by PhungVanHoa** [comment]
any news on this issue?. I have the same problem.

**Comment by dims** [comment]
Closing this issue as it has been open since before January 1, 2024. If this is still relevant, please feel free to reopen or create a new issue.
